File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/UpdateDataLakeCredentialDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The set of details required to update an existing credential object.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UpdateDataLakeCredentialDetails.Builder.class)

public final class UpdateDataLakeCredentialDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"displayName", "credentialDescription", "type", "credentialDetails"})
    public UpdateDataLakeCredentialDetails(String displayName, String credentialDescription, CredentialType type, CredentialDetails credentialDetails) {
        super();
        this.displayName = displayName;
        this.credentialDescription = credentialDescription;
        this.type = type;
        this.credentialDetails = credentialDetails;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

                /**
         * A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.
         * @param displayName the value to set
         * @return this builder
         **/
        

        public Builder displayName(String displayName) {
        this.displayName = displayName;
        return this;
        }
            /**
     * A brief summary of the credential object and its purpose.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("credentialDescription")
        private String credentialDescription;

                /**
         * A brief summary of the credential object and its purpose.
         * @param credentialDescription the value to set
         * @return this builder
         **/
        

        public Builder credentialDescription(String credentialDescription) {
        this.credentialDescription = credentialDescription;
        return this;
        }
            /**
     * The type of credential stored. Allowed values are defined in CredentialType.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("type")
        private CredentialType type;

                /**
         * The type of credential stored. Allowed values are defined in CredentialType.
         * @param type the value to set
         * @return this builder
         **/
        

        public Builder type(CredentialType type) {
        this.type = type;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("credentialDetails")
        private CredentialDetails credentialDetails;

        

        public Builder credentialDetails(CredentialDetails credentialDetails) {
        this.credentialDetails = credentialDetails;
        return this;
        }


        public UpdateDataLakeCredentialDetails build() {
            UpdateDataLakeCredentialDetails model = new UpdateDataLakeCredentialDetails(this.displayName
                    , this.credentialDescription
                    , this.type
                    , this.credentialDetails);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UpdateDataLakeCredentialDetails model) {
                this.displayName(model.getDisplayName());
    this.credentialDescription(model.getCredentialDescription());
    this.type(model.getType());
    this.credentialDetails(model.getCredentialDetails());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * A brief summary of the credential object and its purpose.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("credentialDescription")
    private final String credentialDescription;

        /**
     * A brief summary of the credential object and its purpose.
     * @return the value
     **/
    
    public String getCredentialDescription() {
        return credentialDescription;
    }

    
        /**
     * The type of credential stored. Allowed values are defined in CredentialType.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final CredentialType type;

        /**
     * The type of credential stored. Allowed values are defined in CredentialType.
     * @return the value
     **/
    
    public CredentialType getType() {
        return type;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("credentialDetails")
    private final CredentialDetails credentialDetails;

    
    public CredentialDetails getCredentialDetails() {
        return credentialDetails;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UpdateDataLakeCredentialDetails(");
        sb.append("displayName=").append(String.valueOf(this.displayName));
        sb.append(", credentialDescription=").append(String.valueOf(this.credentialDescription));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", credentialDetails=").append(String.valueOf(this.credentialDetails));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UpdateDataLakeCredentialDetails)) {
            return false;
        }

        UpdateDataLakeCredentialDetails other = (UpdateDataLakeCredentialDetails) o;
        return java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.credentialDescription, other.credentialDescription) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.credentialDetails, other.credentialDetails);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.credentialDescription == null ? 43 : this.credentialDescription.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.credentialDetails == null ? 43 : this.credentialDetails.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/update_data_lake_credential_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UpdateDataLakeCredentialDetails(object):
    """
    The set of details required to update an existing credential object.
    """

    #: A constant which can be used with the type property of a UpdateDataLakeCredentialDetails.
    #: This constant has a value of "SECRET_TOKEN"
    TYPE_SECRET_TOKEN = "SECRET_TOKEN"

    #: A constant which can be used with the type property of a UpdateDataLakeCredentialDetails.
    #: This constant has a value of "VAULT_REFERENCE"
    TYPE_VAULT_REFERENCE = "VAULT_REFERENCE"

    #: A constant which can be used with the type property of a UpdateDataLakeCredentialDetails.
    #: This constant has a value of "SERVICE_ACCOUNT"
    TYPE_SERVICE_ACCOUNT = "SERVICE_ACCOUNT"

    def __init__(self, **kwargs):
        """
        Initializes a new UpdateDataLakeCredentialDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param display_name:
            The value to assign to the display_name property of this UpdateDataLakeCredentialDetails.
        :type display_name: str

        :param credential_description:
            The value to assign to the credential_description property of this UpdateDataLakeCredentialDetails.
        :type credential_description: str

        :param type:
            The value to assign to the type property of this UpdateDataLakeCredentialDetails.
            Allowed values for this property are: "SECRET_TOKEN", "VAULT_REFERENCE", "SERVICE_ACCOUNT"
        :type type: str

        :param credential_details:
            The value to assign to the credential_details property of this UpdateDataLakeCredentialDetails.
        :type credential_details: oci.aidataplatform_dp.models.CredentialDetails

        """
        self.swagger_types = {
            'display_name': 'str',
            'credential_description': 'str',
            'type': 'str',
            'credential_details': 'CredentialDetails'
        }

        self.attribute_map = {
            'display_name': 'displayName',
            'credential_description': 'credentialDescription',
            'type': 'type',
            'credential_details': 'credentialDetails'
        }

        self._display_name = None
        self._credential_description = None
        self._type = None
        self._credential_details = None

    @property
    def display_name(self):
        """
        Gets the display_name of this UpdateDataLakeCredentialDetails.
        A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.


        :return: The display_name of this UpdateDataLakeCredentialDetails.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this UpdateDataLakeCredentialDetails.
        A user-friendly name for the credential object. This value does not have to be unique and can be changed. Must start with a letter and contain only letters, numbers, or underscores. Avoid entering confidential information.


        :param display_name: The display_name of this UpdateDataLakeCredentialDetails.
        :type: str
        """
        self._display_name = display_name

    @property
    def credential_description(self):
        """
        Gets the credential_description of this UpdateDataLakeCredentialDetails.
        A brief summary of the credential object and its purpose.


        :return: The credential_description of this UpdateDataLakeCredentialDetails.
        :rtype: str
        """
        return self._credential_description

    @credential_description.setter
    def credential_description(self, credential_description):
        """
        Sets the credential_description of this UpdateDataLakeCredentialDetails.
        A brief summary of the credential object and its purpose.


        :param credential_description: The credential_description of this UpdateDataLakeCredentialDetails.
        :type: str
        """
        self._credential_description = credential_description

    @property
    def type(self):
        """
        Gets the type of this UpdateDataLakeCredentialDetails.
        The type of credential stored. Allowed values are defined in CredentialType.

        Allowed values for this property are: "SECRET_TOKEN", "VAULT_REFERENCE", "SERVICE_ACCOUNT"


        :return: The type of this UpdateDataLakeCredentialDetails.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this UpdateDataLakeCredentialDetails.
        The type of credential stored. Allowed values are defined in CredentialType.


        :param type: The type of this UpdateDataLakeCredentialDetails.
        :type: str
        """
        allowed_values = ["SECRET_TOKEN", "VAULT_REFERENCE", "SERVICE_ACCOUNT"]
        if not value_allowed_none_or_none_sentinel(type, allowed_values):
            raise ValueError(
                "Invalid value for `type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._type = type

    @property
    def credential_details(self):
        """
        Gets the credential_details of this UpdateDataLakeCredentialDetails.

        :return: The credential_details of this UpdateDataLakeCredentialDetails.
        :rtype: oci.aidataplatform_dp.models.CredentialDetails
        """
        return self._credential_details

    @credential_details.setter
    def credential_details(self, credential_details):
        """
        Sets the credential_details of this UpdateDataLakeCredentialDetails.

        :param credential_details: The credential_details of this UpdateDataLakeCredentialDetails.
        :type: oci.aidataplatform_dp.models.CredentialDetails
        """
        self._credential_details = credential_details

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
